src/Core/src/Extensions/ArrayExtensions.cs

```diff
@@ -64,5 +64,7 @@ public static T[] RemoveAt<T>(this T[] self, int index)
 		/// <param name="self">The array to retrieve the last item from.</param>
 		/// <returns>An object of type <typeparamref name="T"/> that is the last item in the collection.</returns>
 		public static T Last<T>(this T[] self) => self[self.Length - 1];
+
+		internal static T? FirstOrDefaultNoLinq<T>(this T[] items) => items is null || items.Length == 0 ? default : items[0];
```

Comment:
@mandel-macaque any complaints?

Reply:
Looks good, but I am surprise the compiler is not printing a warning with nullability enabled saying something like 'items is not set as nullable'. I think we probably want to do something like:
```suggestion
		internal static T? FirstOrDefaultNoLinq<T>(this T[]? items) => items is null || items.Length == 0 ? default : items[0];
```
Right?